Firefox/sync-data.js

```diff
@@ -0,0 +1,16 @@
+/* globals require, exports */
+let prefs = require('sdk/preferences/service');
+let prefRoot = 'extensions.xkit7.';
+
+exports.setSyncLegacyDataPort = function(port) {
+
+	// Get all and broadcast to webext
+	let existingKeys = prefs.keys(prefRoot);
+	let xkitStorage = {};
+	for (let properName of existingKeys) {
+		let name = properName.substring(prefRoot.length);
+		let value = prefs.get(properName, null);
+		xkitStorage[name] = value;
+	}
+	port.postMessage(xkitStorage);
```

Comment:
My concern is that this is completely synchronous, so it might choke on very large storages (pokes, postarchive, outbox, etc.)